src/ADAL.PCL/HttpClientWrapper.cs

```diff
@@ -129,11 +130,15 @@ public int TimeoutInMilliSeconds
                 {
                     try
                     {
-                        throw new HttpRequestException(string.Format(CultureInfo.CurrentCulture, " Response status code does not indicate success: {0} ({1}).", (int)webResponse.StatusCode, webResponse.StatusCode));
+                        StringBuilder httpErrorResponse = new StringBuilder();
```

Comment:
if webResponse contains string then why is another stream being created just to create another string? This could be simplified to "new exception(webResponse.ResponseString)"